lessons/codap-transformers2/langs/en-us/index.adoc

```diff
@@ -0,0 +1,81 @@
+| Lesson Goals
+| Students will be able to...
+
+* write a contract, purpose statement, and expression
+* use the `Sort` Transformer order the Animals Dataset by a number of criteria
+
+| Student-facing Lesson Goals
+|
+
+* Let’s learn how transform one table into another.
+
+| Materials
+|[.materials-links]
+
+@material-links
+
+|===
+
+== Sorting @duration{25 minutes}
+
+=== Overview
+Students _change a table,_ sorting it by a particular column. Then, students learn a new and more dynamic way of sorting... with the Sort Transformer!
+
+=== Launch
+
+[.lesson-instruction]
+- Open the @starter-file{animals} in CODAP.
+- How many animals' names start with the letter S?
+** _Once you notice students deep in concentration, looking for and tallying animals with S-names, move on to the next question..._
+- What would make it easier to figure out how many animals' names start with the letter S?
+
+Explain to students that, in this lesson, they will learn two ways of sorting. The first way is to actually alter the original table. The second way is to create a transformed copy of the original table, which automatically has the animals sorted.
+
+[.lesson-instruction]
+- When manipulating data, what do you think might be some advantages and disadvantages to changing the actual table?
+** _Advantage: It's quick, easy and intuitive._
+** _Disadvantage: If I want to refer to the previous arrangement of animals for any reason... it might no longer exist._
+- What are some advantages and disadvantages to creating a new modified table?
+** _Advantage: I will be able to refer back to my old table, and maybe even compare the new table with the old table. I can use my Transformer again later once I save it!_
+** _Disadvantage: It will take a bit of typing to set it up, and a few extra clicks._
+
+You will direct students to sort various columns on the table by clicking the attribute name and then selecting from the drop-down menu either `Sort Ascending` or `Sort Descending`. This will modify the original table.
+
+[.lesson-instruction]
+- Let's sort the animals alphabetically by name and see if we get the same answer as we did before! _(The image below shows where to click to sort the animals alphabetically by name.)_
+@centered-image{images/sort-ascending-by-name.png, 450}
+- Can you sort the animals by age, from youngest to oldest?
+- How about from heaviest to lightest?
+- Now sort the animals table by how long it took for each animal to be adopted, in ascending order.
+
+=== Investigate
+
+Now, let's try sorting with a Transformer!
+
+[.lesson-instruction]
+Complete the @printable-exercise{pages/codap-sort-transformer.adoc} worksheet, which will walk you through creating, applying, and saving a `Sort` Transformer.
+
+@right{@image{images/sort-transformer.png,250}}
+
+Initially, students might question the value of using a Transformer to sort. That is okay! The sort functionality embedded into CODAP is a more efficient choice only in some contexts.
```

Comment:
I think this is more of a synthesize question OR workbook page question. We should ask students to think about what those contexts might be!